refactor(core): Removes unused flag for change detection provider check

SkyZeroZx opened the pull request:
… tree shaking

Moves internal toggles from the global scope into functions to avoid DCE bailouts and improve tree shaking

Inspired by https://github.com/angular/angular/pull/64740

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

SkyZeroZx commented on packages/core/src/animation/utils.ts:
```diff
@@ -27,6 +25,7 @@ export const areAnimationSupported =
  * Helper function to check if animations are disabled via injection token
  */
 export function areAnimationsDisabled(lView: LView): boolean {
+  const DEFAULT_ANIMATIONS_DISABLED = false;
```
Done

JeanMeche commented on packages/core/src/linker/destroy_ref.ts:
```diff
@@ -12,8 +12,6 @@ import {LView} from '../render3/interfaces/view';
 import {getLView} from '../render3/state';
 import {removeLViewOnDestroy, storeLViewOnDestroy} from '../render3/util/view_utils';
 
-const EXECUTE_CALLBACK_IF_ALREADY_DESTROYED = false;
```
Can you create a separate PR to remove that flag, cf the TODO on l87.
G3 has been cleaned up.

JeanMeche commented on packages/core/src/platform/bootstrap.ts:
```diff
@@ -28,8 +28,6 @@ import {stringify} from '../util/stringify';
 import {isPromise} from '../util/lang';
 import {PendingTasksInternal} from '../pending_tasks_internal';
 
-const REQUIRE_ONE_CD_PROVIDER_BOOTSTRAP_MODULE = false;
```
My bad this one is still used internally. Let's go with your initial change (moving the flag inside the if block.

SkyZeroZx committed:
refactor(core): move internal toggles inside function scope to enable tree shaking

Moves internal toggles from the global scope into functions to avoid DCE bailouts and improve tree shaking

atscott commented:
All code that uses this flag is already under the dev mode check. moving this inside the function does not help tree shaking. closing.
